test/sockets.test.js

```diff
@@ -1,63 +1,186 @@
 import 'regenerator-runtime/runtime';
 
 import Socket from '../src/sockets';
-import Logger from '../src/logger';
+import { Logger, SOCKET_PING } from 'syft-helpers.js';
 
 import { WebSocket, Server } from 'mock-socket';
 
 global.WebSocket = WebSocket;
 
 const url = 'ws://localhost:8080/';
 
-const onMessage = event => console.log('NEW MESSAGE', event);
-const onOpen = event => console.log('NEW OPEN', event);
-const onClose = event => console.log('NEW CLOSE', event);
-const onError = event => console.log('NEW ERROR', event);
+/**
+ * Creates promise from the first event call
+ * @param emitter Object that emits the event
+ * @param event Event name
+ * @returns {Promise}
+ */
+function makeEventPromise(emitter, event) {
+  let resolver;
+  let promise = new Promise(resolve => resolver = resolve);
+  emitter.on(event, data => resolver(data));
+  return promise;
+}
 
 describe('Sockets', () => {
+  // common test objects
+  let mockServer;
+
   jest.spyOn(console, 'log');
 
+  beforeEach(() => {
+    mockServer = new Server(url);
+    mockServer.connected = makeEventPromise(mockServer, 'connection');
+  });
+
   afterEach(() => {
+    mockServer.close();
     jest.clearAllMocks();
   });
 
-  test('can communicate with a socket server', async () => {
-    const message = { message: 'This is a test' };
+  test('keep-alive messages', async() => {
+    const keepAliveTimeout = 300;
+    const expectedMessagesCount = 3;
 
-    const mockServer = new Server(url);
     const mySocket = new Socket({
       url,
       logger: new Logger(true),
-      onMessage: event => onMessage(event),
-      onOpen: event => onOpen(event),
-      onClose: event => onClose(event),
-      onError: event => onError(event)
+      keepAliveTimeout
     });
+    let serverSocket = await mockServer.connected;
```

Comment:
Again... use `const`, everywhere.  This will be the last comment I write about in my review, but please change all references that can be changed.

Reply:
Fixed